components/brave_wallet_ui/page/screens/swap/adapters.ts

```diff
@@ -296,10 +301,19 @@ export function makeSOLSendTransaction (
   return sendTransaction
 }
 
-export function makeSwitchAccount () {
-  async function switchAccount (account: WalletAccount) {
+export function makeSwitchAccount(accounts: WalletAccountType[]) {
+  async function switchAccount(account: WalletAccount) {
     const { keyringService } = getAPIProxy()
-    await keyringService.setSelectedAccount(account.address, account.coin)
+
+    // TODO(apaymyshev): swap's WalletAccount should somehow provide
+    // corresponding instance of BraveWallet.AccountInfo
+    const keyringId = accounts.find(
+      (acc) => acc.address === account.address
+    )?.keyringId
```

Comment:
Noting here that we'll getting rid of the duplicate swap-specific types as part of the process to bring back swap code to brave-core. This is WIP.